src/Files.App/Actions/Content/Tags/OpenAllTaggedActions.cs

```diff
@@ -33,13 +33,14 @@ public OpenAllTaggedActions()
 
 		public async Task ExecuteAsync()
 		{
-			var files = _tagsContext.TaggedItems.Where(item => !item.isFolder);
-			var folders = _tagsContext.TaggedItems.Where(item => item.isFolder);
-
-			await Task.WhenAll(files.Select(file 
-				=> NavigationHelpers.OpenPath(file.path, _pageContext.ShellPage!)));
-
-			folders.ForEach(async folder => await NavigationHelpers.OpenPathInNewTab(folder.path, false));
+			var filePaths = _tagsContext.TaggedItems.Where(item => !item.isFolder).Select(f => f.path);
+			var folderPaths = _tagsContext.TaggedItems.Where(item => item.isFolder).Select(f => f.path);
+
+			foreach (var path in filePaths) 
+				await NavigationHelpers.OpenPath(path, _pageContext.ShellPage!);
```

Comment:
I don't think the order is important here and `Task.WhenAll` should be used.

Reply:
https://github.com/files-community/Files/pull/15005#discussion_r1545862037

Reply:
I think yes for folders, but no for files, because they are opened by other applications.

Reply:
Good point